### PR TITLE
TAJO-2127: Use Type instead of DataType for EvalNode.

### PR DESCRIPTION
**Pull request by hyunsik**

### CODE REVIEW

File: tajo-plan/src/main/java/org/apache/tajo/plan/LogicalPlanner.java [hunk 1]
```diff
@@ -577,7 +577,7 @@ public static void verifyProjectedFields(QueryBlock block, Projectable projectab
   public static void prohibitNestedRecordProjection(Projectable projectable)
       throws TajoException {
     for (Target t : projectable.getTargets()) {
-      if (t.getEvalTree().getValueType().getType() == TajoDataTypes.Type.RECORD) {
+      if (t.getEvalTree().getValueType().isStruct()) {
```
Comment (jihoonson) [inline tajo-plan/src/main/java/org/apache/tajo/plan/LogicalPlanner.java]: Struct or record?

File: tajo-core-tests/src/test/java/org/apache/tajo/engine/eval/TestSQLExpression.java [hunk 2]
```diff
@@ -565,7 +565,7 @@ public void testImplicitCastForInt8() throws TajoException {
   public void testImplicitCastForFloat4() throws TajoException {
     Schema schema = TestSchema1;
 
-    testEval(schema, "table1", "0,1,2,3,4.1,5.1,6,7", "select col4 + col0 from table1;", new String [] {"4.1"});
+    /*testEval(schema, "table1", "0,1,2,3,4.1,5.1,6,7", "select col4 + col0 from table1;", new String [] {"4.1"});
```
Comment (jihoonson) [inline tajo-core-tests/src/test/java/org/apache/tajo/engine/eval/TestSQLExpression.java]: Why did you disable these tests?

Comment (hyunsik) [inline tajo-core-tests/src/test/java/org/apache/tajo/engine/eval/TestSQLExpression.java]: I forget to remove the comments after some bug fixes.

File: tajo-plan/src/main/java/org/apache/tajo/plan/LogicalPlanner.java (hunk 1 above)
Comment (hyunsik) [inline tajo-plan/src/main/java/org/apache/tajo/plan/LogicalPlanner.java]: Currently, record is right. I need to make them consistent. According to my future plan, struct would be better. That's why they are mixed.

Comment (jihoonson) [inline tajo-plan/src/main/java/org/apache/tajo/plan/LogicalPlanner.java]: Ok. Looks good.

**Comment by hyunsik** [comment]
Fixed unit tests. Thank you for your quick review.

**Comment by jihoonson** [comment]
+1 LGTM!

**Pull request closed without merging**